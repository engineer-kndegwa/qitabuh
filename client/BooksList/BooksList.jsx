import React from 'react';
import PropTypes from 'prop-types';

// Components
import Book from "../Book/Book";

const BooksList = ({books, selectBook, selectDuration}) => {
    if(!books || (books && books.length === 0)){
        return null
    }

    const booksHtml = books.map(book=>{
        return <Book 
        key={book.id} 
        {...book} 
        selectBook={selectBook} 
        selectDuration={selectDuration}
        />
    });
    return(
        <ul className="books-list-ctn">
            {booksHtml}
        </ul>
    )
};

BooksList.propTypes = {
    books: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            isSelected: PropTypes.bool.isRequired,
            category: PropTypes.string,
            durationInDays: PropTypes.number.isRequired
        })
    ),
    selectBook: PropTypes.func.isRequired,
    selectDuration: PropTypes.func.isRequired
}

export default BooksList;