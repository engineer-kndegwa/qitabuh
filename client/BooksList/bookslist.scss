.books-list-ctn {
    width: 80%;
    margin: 0 auto;
    list-style-type: none;
    height: 60vh;
    overflow: scroll;
    border-radius: 4px;
    padding: 0 3em;
}