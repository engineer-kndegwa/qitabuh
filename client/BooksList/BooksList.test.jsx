import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

// Component
import BooksList from "./BooksList";

const dummyProps = {};

beforeAll(()=>{
    const props = {
        books: [],
        selectBook: ()=>{},
        selectDuration: ()=>{}
    };

    const propsAlt = {
        books: [{
            id: 'abcd',
            title: "Title",
            durationInDays: 0,
            isSelected: false,
            category: ""
        }],

        selectBook: ()=>{},
        selectDuration: ()=>{}
    };

    dummyProps.props = props;
    dummyProps.propsAlt = propsAlt;
    return dummyProps;
});

test('Should render correctly when all props are passed.', () => {
    const { props, propsAlt } = dummyProps;
    const tree = renderer.create(<BooksList {...props} />).toJSON();
    const treeAlt = renderer.create(<BooksList {...propsAlt} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(treeAlt).toMatchSnapshot();
});

test('Should render null when there are no items', () => {
    const { props } = dummyProps;
    const wrapper = shallow(<BooksList {...props}/>);
    expect(wrapper.type()).toEqual(null);
});
