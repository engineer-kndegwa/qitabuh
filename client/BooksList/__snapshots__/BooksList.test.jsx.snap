// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly when all props are passed. 1`] = `null`;

exports[`Should render correctly when all props are passed. 2`] = `
<ul
  className="books-list-ctn"
>
  <li
    className="book-ctn"
    onClick={[Function]}
  >
    <label
      className="checkbox-ctn"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkbox-text"
      >
        Title
      </span>
    </label>
  </li>
</ul>
`;
